tests/test_inference.py

```diff
@@ -1,18 +1,135 @@
+import os
+import random
+
+import pytest
+import torch
+
 from cities.modeling.model_interactions import InteractionsModel
+from cities.queries.causal_insight import CausalInsight
+from cities.utils.cleaning_utils import find_repo_root
+from cities.utils.data_grabber import list_interventions, list_outcomes
+
+os.environ["CUDA_VISIBLE_DEVICES"] = ""
+device = torch.device("cpu")
+
+root = find_repo_root()
+
+interventions = list_interventions()
+outcomes = list_outcomes()
+shifts = [1, 2, 3]
 
+# TODO add list_fips() function
+# and parametrize with random fips
+# and uniformly random intervention value
+fips = 1003
+intervened_value = 0.9
 
-# TODO this needs to be parametrized by outcome datasets,
-# intervention datasets and forward shifts
-def test_InteractionsModel():
+# running for all is too inefficient
+# let's just run for random on a regular basis
+# comment out for a full battery of tests
+# intervention = [random.choice(interventions)]
+# outcome = [random.choice(outcomes)]
+
+interventions = [random.choice(interventions)]
```

Comment:
I'm not sure what the pytest-standard implementation for this is, but consider seeding this randomness so that tests consistently fail/succeed.

Reply:
I have considered this and decided against it. The stability-optimal version would be to run this for all combinations. A compute-minimal way would be to fix one particular setting, which setting the seed would effectively do. I want a middle path, where randomness goes through various combinations every now and then, so that if some particular setting is a source of a problem there is a non-negligible chance of it being eventually discovered. Identifying which combination is a problem is not a problem as it is enough to look at the log/output.